### DISCUSSION TITLE
securitymodule_01_colony, bot attempts to follow into cell, but can't open the hatch

### DISCUSSION THREAD

**Discussion by kumilanka** [FakeFishGames/Barotrauma#15215] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Bot keeps attempting to follow down the ladder into jail cell, can't open hatch.

### Reproduction steps

1. open securitymodule_01_colony in sub editor
2. enter play mode
3. bring self and one bot to module
4. teleport into jail cell
5. order bot to follow
6. bot will attempt to climb ladder and spaz up and down endlessly

![image](https://github.com/user-attachments/assets/aa2cd261-4427-4ad9-880c-419cc42e3aed)

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by itchyOwl** [comment]
Quickly tested, and it seems you shouldn't probably be able to access the cells anyway? The door refuses to open. The bots probably don't realize this, because technically they have access to the door (didn't confirm this by debugging).

So, I'd say this could be a design issue or non-issue in the first place?

**Reply by Regalis11** [reply]
What's happening here is there is a button that opens the door that the bots can and do use, but there's also an additional circuit that forces the door to close (and the bot doesn't realize that and keeps attempting to open the door and get past it). 

We could potentially address that by modifying the circuit somehow, but imo we should expect there to sometimes be more complicated door opening mechanisms that fail to open the door despite there being a button that seems like it should open the door. So if we wanted to address this, maybe we could do something like making the bots give up on opening the door after a while if it refuses to open.
